### PR TITLE
fixes #17504, #15619 - use generated puppet.conf when running Puppet

### PR DESCRIPTION
**Pull request by domcleal**

### CODE REVIEW

File: lib/kafo/kafo_configure.rb [hunk 1]
```diff
@@ -411,23 +412,26 @@ def run_installation
       hiera.write_configs
       self.class.exit_handler.register_cleanup_path(hiera.temp_dir)
 
+      puppetconf = PuppetConfigurer.new(
+        'color'        => false,
+        'evaltrace'    => !!@progress_bar,
+        'hiera_config' => hiera.config_path,
+        'noop'         => !!noop?,
+        'profile'      => !!profile?,
+        'show_diff'    => false,
```
Comment (ares) [inline lib/kafo/kafo_configure.rb]: based on jenkins finding, trailing comma does not work in Ruby 1.8

File: lib/kafo/puppet_configurer.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    end
+
+    def []=(key, value)
+      @settings[key] = value
+    end
+
+    def write_config
+      @logger.debug("Writing Puppet config file at #{@temp_file.path}")
+      @temp_file.open
+      @temp_file.truncate(0)
```
Comment (ares) [inline lib/kafo/puppet_configurer.rb]: just out of curiosity, is this because of race condition even when using `Tempfile`?

**Comment by ares** [comment]
Except the 1.8 issue it looks good.

File: lib/kafo/puppet_configurer.rb (hunk 2 above)
Comment (domcleal) [inline lib/kafo/puppet_configurer.rb]: It's in case the method is called twice so the file gets overwritten fully.

**Comment by domcleal** [comment]
PR updated with the trailing comma removed.

**Comment by ares** [comment]
Thanks @domcleal, merging.

**Pull request merged**